addons/sale_timesheet/static/tests/so_line_field_tests.js

```diff
@@ -0,0 +1,136 @@
+/* @odoo-module */
+
+import { startServer } from "@bus/../tests/helpers/mock_python_environment";
+import { addModelNamesToFetch } from "@bus/../tests/helpers/model_definitions_helpers";
+
+import { start } from "@mail/../tests/helpers/test_utils";
+
+import { click, editInput, getFixture, triggerEvent } from "@web/../tests/helpers/utils";
+import { contains } from "@web/../tests/utils";
+import { setupViewRegistries } from "@web/../tests/views/helpers";
+
+addModelNamesToFetch(["sale.order.line", "account.analytic.line"]);
+
+let serverData;
+let target;
+
+QUnit.module("Sale Order Line Field Tests", (hooks) => {
+    hooks.beforeEach(async () => {
+        const pyEnv = await startServer();
+        const so_line = pyEnv["sale.order.line"].create([{ name: "Sale Order Line 1" }]);
+        pyEnv["sale.order.line"].create([{ name: "Sale Order Line 2" }]);
+        pyEnv["account.analytic.line"].create([
+            {
+                so_line: so_line,
+            },
+        ]);
+        serverData = {
+            views: {
+                "account.analytic.line,false,form": `<form>
+                        <field name="so_line" widget="so_line_field"/>
+                    </form>`,
+                "account.analytic.line,false,list": `<tree editable="bottom">
+                        <field name="so_line" widget="so_line_field"/>
+                    </tree>`,
+                "project.task,false,form": `<form>
+                    <field name="timesheet_ids">
+                        <tree editable="bottom">
+                            <field name="so_line" widget="so_line_field"/>
+                            <field name="is_so_line_edited" column_invisible="True"/>
+                        </tree>
+                    </field>
+                </form>`,
+            },
+        };
+        target = getFixture();
+        setupViewRegistries();
+    });
+
+    QUnit.test(
+        "Check whether so_line_field widget works as intended in form view",
+        async function (assert) {
```

Comment:
I think its better to use assert.expect(1) here instead of the assert.step. The step is usually used to ensure that a flow is done in a certain order. Here, you're just using it to ensure that the mockrpc is correctly called. The assert.expect fulfill that role, and its 1 line shorter :smile: 

```js
assert.expect(1);
```

Reply:
I agree with you :)
Here I think I have to use assert.expect(2) instead of assert.expect(1) because there is a call to contains() which performs an assertion.